Modules/socketmodule.c

```diff
@@ -789,6 +789,8 @@ internal_select(PySocketSockObject *s, int writing, _PyTime_t interval,
     ms = _PyTime_AsMilliseconds(interval, _PyTime_ROUND_CEILING);
     assert(ms <= INT_MAX);
 
+    ms = ms < 0 ? -1 : ms;
```

Comment:
Please add a comment explaining why you do that.

I would prefer a regular if to make it more explicit that the value is only modified it's negative. Like:

```
if (ms < -1) {
    /* FreeBSD implementation of poll(2) restricts timeout argument to be either zero, or positive, or equal to INFTIM (-1). */
    ms = -1;
}
```

Reply:
Sure thing, see the updates. I borrowed the wording from #4033 -- the author was tackling exactly the same `poll(2)` problem.

Also I have created a news entry. I am not sure whether there's necessity, though.